### PR TITLE
Adding geolocation button

### PR DESCRIPTION
**Pull request by pakb**
[Test link](https://web-mapviewer.dev.bgdi.ch/feature_BGDIINF_SB-1437_geolocation/index.html)

### CODE REVIEW

**Comment by davidoesch** [comment]
shall I give feedback? ready?

**Comment by pakb** [comment]
The first geolocation and error handling is implemented, but I can't get the "tracking" to work yet.
You can already give a look at the UI element and how it react to not have the Geolocation authorization and such, tracking will come later today or tomorrow.

**Comment by davidoesch** [comment]
- [ ] when hitting the gelocation button: zoom level is z=14.5 , expected result 15.5 ( corresponds to mf-geoadmin3 and gmaps app: user actually gets at this scale where he stands) or is there another reason? Even if there is a weak gsp signal (huge red circle ) we stay at this z

**Comment by davidoesch** [comment]
otherwise I tested so far on iphone 7 with iOS 14.2 - : and it looks perfect. (not tested on other devices etc like desktop not on android etc.)

thx Patrick!

**Comment by pakb** [comment]
oh right, I miss-read the zoom level of the current viewer (zoom LV95 : 8) which correspond to the 1:25'000 map, that's fixed. Tracking should now be operational, I'll wait for the rain to settle to try it out outside later today

File: src/modules/store/plugins/geolocation-management.plugin.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        if (mutation.type === 'setGeolocationActive') {
+            if (state.geolocation.active) {
+                navigator.geolocation.getCurrentPosition(
+                    position => {
+                        // if geoloc was previously denied, we clear the flag
+                        if (state.geolocation.denied) {
+                            store.dispatch('setGeolocationDenied', false);
+                        }
+                        // we center the view on the position of the user
+                        store.dispatch('setCenter', readPositionEpsg3857(position));
```
Comment (pakb) [inline src/modules/store/plugins/geolocation-management.plugin.js:55]: that is unnecessary, I should remove it (it's already done by `handlePositionAndDispatchToStore`

**Comment by davidoesch** [comment]
and: I like your idea to have the button with a transparency

File: src/api/layers.api.js [hunk 2]
```diff
@@ -94,19 +102,39 @@ export class WMTSLayer extends Layer {
         if (this.timeSeriesConfig) {
             timestamp = this.timeSeriesConfig.currentTimestamp
         }
-        return `${WMTS_BASE_URL}1.0.0/${this.id}/default/${timestamp}/3857/{z}/{x}/{y}.${this.format}`;
+        return `${this.baseURL}1.0.0/${this.id}/default/${timestamp}/3857/{z}/{x}/{y}.${this.format}`;
+    }
+
+    getURLs() {
```
Comment (boecklic) [inline src/api/layers.api.js:108]: maybe add a comment about what this function does and why it's used.

File: src/modules/store/plugins/geolocation-management.plugin.js (hunk 1 above)
Comment (boecklic) [reply src/modules/store/plugins/geolocation-management.plugin.js:55]: So if I understand correctly, geolocation is relying on the same handling as when position is entered via URL params?

**Review by boecklic — APPROVED** [review]

**Comment by davidoesch** [comment]
Maybe this helps:  There is an url Parameter called geolocation=true which activates geolocation

**Comment by pakb** [comment]
@boecklic yes, to center the view on the user's geolocation we go through the same mechanism as if we add written a new location in the URL (all those `set where the map look` functions goes through the same endpoint in the store in the end, that's the goal)

**Pull request merged**